src/lib/state/local/index.ts

```diff
@@ -0,0 +1,32 @@
+import { EventListener } from "../../events";
+import { FunctionUtil } from "../../util/functions";
+import { StateSetter } from "../types";
+import { LocalStateImplementation } from "./types";
+
+export class LocalState<T = unknown> implements LocalStateImplementation<T> {
+	private readonly subscription = new EventListener<[T]>();
+
+	private value: T;
+
+	constructor(initialValue: StateSetter<T>) {
+		this.value = FunctionUtil.isFunction(initialValue) ? initialValue() : initialValue;
+	}
+
+	public subscribe(): EventListener<[T]> {
+		return this.subscription;
+	}
+
+	public set(setter: StateSetter<T>): void {
+		if (FunctionUtil.isFunction(setter)) {
+			this.value = setter(this.value);
```

Comment:
Do we want this to be immutable for tables? If T is a Map here then I think it would make more sense to simply add the values you want to the Map rather than having to copy the contents and then add.

While we can't detect for Maps and Sets at runtime, maybe a type check that says if T extends Map or Set then void is valid return type. I think this might be an issue from my original code actually 😅

Reply:
Pretty sure having a `Map` shouldn't be possible.

Reply:
ur rite